go/libraries/doltcore/sqle/expranalysis/expranalysis.go

```diff
@@ -92,16 +98,26 @@ func parseCreateTable(ctx *sql.Context, tableName string, sch schema.Schema) (*p
 
 	query := createTable
 
-	mockDatabase := memory.NewDatabase("mydb")
-	mockProvider := memory.NewDBProvider(mockDatabase)
-	catalog := analyzer.NewCatalog(mockProvider)
-	catalog.AuthHandler = sql.NoopAuthorizationHandler{}
-
-	// We need a new context for this operation
-	parseCtx := sql.NewEmptyContext()
-	parseCtx.SetCurrentDatabase("mydb")
+	// Dolt creates a new provider (is that necessary?), while Doltgres must use the existing provider, so we split the
+	// logic here depending on whether this is Dolt or Doltgres.
```

Comment:
Interesting! Creating a new provider doesn't seem like it should be necessary. Did you already try using the same code for Dolt as you added for Doltgres? If we could access the DoltDatabaseProvider and use it here, it seems like it might work.

Reply:
I should have updated the comment a little later, but this is called with a variety of sessions, so the previous implementation was there so that it works in all cases. I'll update it so that we're only checking if we have a Dolt session object (Doltgres should _always_ have a Dolt session AFAICT).